pom.xml

```diff
@@ -705,7 +706,7 @@
       <dependency>
         <groupId>org.apache.curator</groupId>
         <artifactId>curator-recipes</artifactId>
-        <version>2.4.0</version>
+        <version>${curator.version}</version>
```

Comment:
that's what I meantioned as an extended patch, I do think it's the right way to do it.

I've just pushed up a a new commit which declared the version of `curator-framework` and `curator-client`, so they will always be consistent and under the control of the spark build. It doesn't scope the new JARs, on the basis that this POM is only setting the versions. They could be given the hadoop.deps.scope though